patterns/grok-patterns

```diff
@@ -1,6 +1,6 @@
 USERNAME [a-zA-Z0-9._-]+
 USER %{USERNAME}
-EMAILLOCALPART [a-zA-Z][a-zA-Z0-9_.+-=:]+
+EMAILLOCALPART [a-zA-Z0-9][a-zA-Z0-9_.+-=:]+
```

Comment:
The non-quoted form of an e-mail address's local part from RFC5322 (`dot-atom`) without backtracking:

```suggestion
EMAILLOCALPART [a-zA-Z0-9!#$%&'*+\-/=?^_`{|}~]+(?:\.[a-zA-Z0-9!#$%&'*+\-/=?^_`{|}~]+)*
```

Valid e-mail addresses it catches that our current pattern doesn't:

 - `a@example.com`
 - `a{b}@example.com`

Cases it does _not_ catch (correctly) that our current pattern does (incorrectly):

 -  `a..b@example.com`
 - `a:b@example.com`

Reply:
Thanks, I know it's imperfect - specifically tried to avoid going too far to just keep changes 'small' for the moment.
The simple change was present in 3 distinct PRs already - no one seems to have cared taking it further, yet :wink: 
Anyhow point taken. This will need some more work if it's not longer just adding specs over another one's commit.